include/cpr/error.h

```diff
@@ -89,6 +90,77 @@ enum class ErrorCode {
     UNKNOWN_ERROR = 1000,
 };
 
+constexpr std::array<std::pair<ErrorCode, const char*>, 63> error_code_to_string_mapping = {{{ErrorCode::OK, "OK"},
```

Comment:
I personally prefer using a `std::unordered_map` here since it is a bit more tolerant and robust against error codes not being in order.

Reply:
you can't make it a constexpr though. I also would prefer that approach.

Reply:
What is the value in making a constexpr version of `to_string` compared to a runtime version?
I agree it makes sense to have a constexpr version, but in my eyes this `to_string` functionality is more helpful if you get an error during runtime e.g. via a response object and then you want to print the error in a useful manner.

Reply:
I have moved to using `inline` instead of constexpr to avoid "duplicate definition" errors on compiling. I will need a more experienced c++ person to tell me if that is an inefficient thing to do. But now we have the map ;)

Reply:
Your approach is good. I'd expect the compiler to implicitly inline the map if needed, but better to make it explicit.